File: monitoring/src/Monitoring.Domain/Gedeeld/AggregateRoot.cs
namespace Monitoring.Domain.Gedeeld;

public abstract class AggregateRoot
{
    private readonly List<IDomainEvent> _events = new();

    protected void RegistreerEvent(IDomainEvent domeinEvent) => _events.Add(domeinEvent);

    public IReadOnlyList<IDomainEvent> TrekEventsLeeg()
    {
        var uit = _events.ToList();
        _events.Clear();
        return uit;
    }
}

File: contract/src/domain/gedeeld/aggregate-root.ts
import type { ContractDomainEvent } from './domain-events.js';

export abstract class AggregateRoot {
  private events: ContractDomainEvent[] = [];

  protected registreerEvent(event: ContractDomainEvent): void {
    this.events.push(event);
  }

  trekEventsLeeg(): ContractDomainEvent[] {
    const uit = this.events;
    this.events = [];
    return uit;
  }
}
